Remove obsolete app checks from shouldUseMouseGestureRecognizer

annevk opened the pull request:
#### cea182dea916491025be8eeab9a3223b0e3c4c1b
<pre>
Remove obsolete app checks from shouldUseMouseGestureRecognizer
<a href="https://bugs.webkit.org/show_bug.cgi?id=293353">https://bugs.webkit.org/show_bug.cgi?id=293353</a>
<a href="https://rdar.apple.com/151973434">rdar://151973434</a>

Reviewed by Wenson Hsieh.

The combination of linked-on-or-after, mandatory usage of newer SDKs,
and most of these apps getting active updates, enables us to simplify
our code.

This code was added back in 2020 in 222230@main and 228838@main.

Canonical link: <a href="https://commits.webkit.org/296512@main">https://commits.webkit.org/296512@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c19731ded8ca8c3b335a86f94ea53cb29e742aa0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108489 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28150 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18574 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113698 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58892 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28839 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36702 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82392 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111437 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22879 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97721 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62828 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22296 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15859 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58417 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/101048 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92250 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15912 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116819 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/107048 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35543 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26195 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91416 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35916 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93995 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91217 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23286 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36113 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13876 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31291 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35445 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40980 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/131331 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35154 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35631 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38506 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36836 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/37ca0c4f0af48336ad1e5d8277eebcfec1857955. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45693

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/04c139907b67f2c12f66b168497f574744013164. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45693

brentfulgham commented on Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
```diff
@@ -568,66 +564,6 @@ static bool applicationBundleIsEqualTo(const String& bundleIdentifierString)
-bool IOSApplication::isEssentialSkeleton()
-{
-    static bool isEssentialSkeleton = applicationBundleIsEqualTo("com.3d4medical.EssentialSkeleton"_s);
-    return isEssentialSkeleton;
```
Unfortunately, this app has not been updated in many years and still requires this quirk. Can you revise the patch to leave this single app in place?

brentfulgham reviewed (changes requested):
This patch is great, except that we can't remove the "Essential Skeleton" quirk because that developer has not updated to a recent SDK. The app still depends on this old behavior. :-(

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c19731ded8ca8c3b335a86f94ea53cb29e742aa0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45693

annevk committed:
Remove obsolete app checks from shouldUseMouseGestureRecognizer
https://bugs.webkit.org/show_bug.cgi?id=293353
rdar://151973434

Reviewed by Wenson Hsieh.

The combination of linked-on-or-after, mandatory usage of newer SDKs,
and most of these apps getting active updates, enables us to simplify
our code.

This code was added back in 2020 in 222230@main and 228838@main.

Canonical link: https://commits.webkit.org/296512@main
